### PR TITLE
fix(agw): Added check at delete session response to send TAU accept if eps_bearer_status flag is set to inactive for default bearer

### PR DESCRIPTION
**Pull request by rsarwad**
…

fix(agw): Added check at delete session response to send TAU accept if eps_bearer_status flag is set to inactive for default bearer
## Summary
Issue1: Flag, ue_context->tau_accept_eps_ber_cntx_status was reset when MME sends TAU accept; On failing to send TAU accept message; this flag was not reset
Resolution: Modified code to reset this flag while we delete the TAU procedure, this was taken care in PR; https://github.com/magma/magma/pull/11795. 
Issue2: With importing this PR on branch v1.7 still there was an issue with mme restart. Consider this flag is set and same is read from Redis DB and while handling attach request; it deletes the previous contexts. While handling Delete session response; we check for flag ue_context->tau_accept_eps_ber_cntx_status and proceed to delete the pdn contexts and decrement the nb_delete_sessions;
Resolution2: Added check at Delete session response to check ue_context->tau_accept_eps_ber_cntx_status and also check nb_delete_sessions

## Test Plan
Executed s1ap sanity test suite and unit test cases.
Verified the changes on both Terra VM setups, AGW: 192.168.60.19 and AGW: 10.22.162.130.

### CODE REVIEW

File: lte/gateway/c/core/oai/tasks/mme_app/mme_app_bearer.c [hunk 1]
```diff
@@ -4762,7 +4763,9 @@ void mme_app_handle_mme_init_local_deactivation(
       ue_context_p, false, bearer_deactv_req_p->ebi,
       bearer_deactv_req_p->no_of_bearers, pdn_context->s_gw_teid_s11_s4,
       REQUEST_ACCEPTED, pdn_context->route_s11_messages_to_s8_task, true);
-  ue_context_p->nb_delete_bearer_cmd--;
+  if (ue_context_p->nb_delete_bearer_cmd) {
```
Comment (ssanadhya) [inline lte/gateway/c/core/oai/tasks/mme_app/mme_app_bearer.c]: Use explicit check for non-zero value. Further, shouldn't this check  happen before we reach this point in code?

```suggestion
  if (ue_context_p->nb_delete_bearer_cmd > 0) {
```

File: lte/gateway/c/core/oai/tasks/mme_app/mme_app_bearer.c [hunk 2]
```diff
@@ -985,7 +985,8 @@ void mme_app_handle_delete_session_rsp(
     OAILOG_FUNC_OUT(LOG_MME_APP);
   }
 
-  if (ue_context_p->tau_accept_eps_ber_cntx_status) {
+  if (ue_context_p->tau_accept_eps_ber_cntx_status &&
+      ue_context_p->nb_delete_sessions) {
```
Comment (ssanadhya) [inline lte/gateway/c/core/oai/tasks/mme_app/mme_app_bearer.c]: Explicit check for no-zero value. Shouldn't the check happen earlier in this function call?
```suggestion
      ue_context_p->nb_delete_sessions > 0) {
```

**Review by ssanadhya — CHANGES_REQUESTED** [review]

File: lte/gateway/c/core/oai/tasks/mme_app/mme_app_bearer.c [hunk 3]
```diff
@@ -985,7 +985,8 @@ void mme_app_handle_delete_session_rsp(
     OAILOG_FUNC_OUT(LOG_MME_APP);
   }
 
-  if (ue_context_p->tau_accept_eps_ber_cntx_status) {
+  if ((ue_context_p->tau_accept_eps_ber_cntx_status) &&
```
Comment (maxhbr) [inline lte/gateway/c/core/oai/tasks/mme_app/mme_app_bearer.c]:
```suggestion
  if (ue_context_p->tau_accept_eps_ber_cntx_status &&
```

**Review by ssanadhya — APPROVED** [review]
lgtm

**Pull request merged**